### ISSUE TITLE
Can we have .dmg file on releases download page?

### ISSUE THREAD

**Issue by juniorb2ss** [Sequel-Ace/Sequel-Ace#58]
That can be good to create a cask formule for `brew`, like we have today for `sequel-pro`: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/sequel-pro.rb

**Comment by Kaspik** [comment]
Could you please hint me an idea or reason why would you prefer Brew instead of Mac AppStore? :)

Thanks!

**Comment by juniorb2ss** [comment]
> Could you please hint me an idea or reason why would you prefer Brew instead of Mac AppStore? :)
> 
> Thanks!

Hello @Kaspik 

Just my preference. Because with brew I have `brew bundle dump` which helps a lot when I change my computer.

I don't see that the possibility of using the brew will remove the use of the Mac AppStore. I particularly find the brew much more dynamic and practical, which makes it much easier. So, just as the sequel-pro allows installation using the brew, I believe that this project can also allow it.

**Comment by jpmurray** [comment]
To add my voice to this, I have a .dotfile install setup that automates the installation process of my work computer, having a cask makes it a breeze.

(Also, but that is just personnal, I haven't logged to the app store in _years_ and I'd like for that to continue! 🙂)

**Comment by Jason-Morcos** [comment]
Would be cool! I know nothing about publishing to homebrew. What do we need to do exactly to make Sequel Ace available as a cask?

**Comment by juniorb2ss** [comment]
> Would be cool! I know nothing about publishing to homebrew. What do we need to do exactly to make Sequel Ace available as a cask?

@Jason-Morcos 

Only .dmg file and `sha256` to make security verification. The rest we can use the same cask used for sequel pro: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/sequel-pro.rb

If we have a .dmg file available, I can create the cask file instructions

**Comment by Kaspik** [comment]
I would suggest including the dmg file here on Github in release - but we will need to figure out code signing first.

Let's investigate this. :)

**Comment by juniorb2ss** [comment]
FYI @Jason-Morcos https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#adding-a-cask

**Comment by jeffbyrnes** [comment]
It’s possible to install Mac App Store stuff via Homebrew:

```bash
brew install mas-cli/tap/mas
# mas install $app_store_id
mas install 1518036000 # Sequel Ace
```

**Comment by Kaspik** [comment]
Ohh love it! Let me create a PR with updated readme as a supported way of installation. :)

**Comment by jpmurray** [comment]
@jeffbyrnes do we need to be logged in to the App Store for that to work ? Because on my end, one of the joy of using homebrew is not to use the app store entirely when possible... ;-)

**Comment by jeffbyrnes** [comment]
You do have to be logged in, yes.

**Comment by jeffbyrnes** [comment]
I’d be happy to help automate non-MAS release going to Homebrew, but this is a good workaround for now!

**Comment by Kaspik** [comment]
Agree, let's add it to the Readme anyways, but keep this open for future reference completely without AppStore.

**Comment by emjayess** [comment]
Many thanks to everyone with the initiative to breathe life into this project! 

I'll echo the sentiments of others here... I love using mac homebrew to manage installed software packages, and loathe using the mac app store. In fact, if an app isn't available via brew, the likelihood that I'll even bother installing it drops by probably 75%.

**Comment by jpmurray** [comment]
Take this a grain of salt, I'm reading and learning about punlishing cask as we go but it seems we could forgot (at least for now!) the usage of an SHA check for the casks. [Taken from their docs page for adding a cask, you can see an example without it from a package called "advancedcolors"](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md):

> And here is one for advancedcolors. Note that it has an unversioned download (the download url does not contain the version number, unlike the example above). It also suppresses the checksum with sha256 :no_check (necessary since the checksum will change when a new distribution is made available). This combination of version :latest and sha256 :no_check is currently the preferred mechanism when a versioned download URL is not available and the cask does not have an appcast.

```
cask 'advancedcolors' do
  version :latest
  sha256 :no_check

  url 'https://advancedcolors.com/AdvancedColors.zip'
  name 'Advanced Colors'
  homepage 'https://advancedcolors.com/'

  app 'AdvancedColors.app'
end
```

Since you said you could help do it, @jeffbyrnes, do you think it's a good avenue ?

**Comment by jeffbyrnes** [comment]
I can certainly take a look, maybe this weekend. This would be my first time defining a cask, but I’ve worked with regular formulæ a few times, so I don’t foresee it being _too_ difficult (famous last words?)

The other approach I’ve seen is to have some kind of CI functionality that handles updating the formula (GitHub Action would work well, I think).

**Comment by Kaspik** [comment]
I'm planning to introduce Github CI here, so if you can wait, we can get back to it.

**Comment by zylo47** [comment]
I'm logged into the app store and when I try the commands provided to install with homebrew above I get the following message:

Redownload Unavailable with This Apple ID
This redownload is not available for this Apple ID either because it was bought by a different user or the item was refunded or cancelled.

**Comment by jeffbyrnes** [comment]
@zylo47 that’s an App Store error, not a Homebrew error. The error says exactly what it means; I would try deleting your existing local copy of the app, logging out & back in to the store, and trying again.

**Comment by Bytesmiths** [comment]
> Could you please hint me an idea or reason why would you prefer Brew instead of Mac AppStore? :)

Well, Homebrew is open source and free, whereas Apple Store is proprietary, and Apple reserves the right to market your personal information. **I have cancelled my Apple Store account, and will not use it.**

**Comment by loranger** [comment]
I'd be glad to have an alternative download solution too, as I cannot install Sequel Ace from the french App Store :

<img width="566" alt="mas install" src="https://user-images.githubusercontent.com/6014/86572804-bbbb9b00-bf73-11ea-8cef-b5dd40cebf47.png">

**Comment by Jason-Morcos** [comment]
> I'd be glad to have an alternative download solution too, as I cannot install Sequel Ace from the french App Store :
> 
> <img alt="mas install" width="566" src="https://user-images.githubusercontent.com/6014/86572804-bbbb9b00-bf73-11ea-8cef-b5dd40cebf47.png">

@loranger You can download the release zip directly from GitHub. We upload a zip for each release.
https://github.com/Sequel-Ace/Sequel-Ace/releases

**Comment by njvack** [comment]
Maybe a weird question here, but: What's the motivation for releasing through the MAS at all? Is an iPad version in the plans? Or a paid version?

**Comment by Jason-Morcos** [comment]
> Maybe a weird question here, but: What's the motivation for releasing through the MAS at all? Is an iPad version in the plans? Or a paid version?

Ease of distribution, no requirement to run any special build server (the demise of Sequel Pro), trusted distribution platform, available on all Macs. No plans for iPad or paid version. Also, why not? If we're gonna notarize and sandbox for security and interoperability (and we follow Apple's rules), why not distribute on MAS?

**Comment by samuelbradshaw** [comment]
As a user, I like using the App Store because it's easier to search for apps and compare them to each other, with standardized rating/reviews, etc. I also like that Apple keeps track of App Store apps I've installed, so when I switch to a different computer, I can download many of the apps I need all in one place. To me, the convenience of having most apps in one place is similar to the convenience of having a package manager for Python libraries, or knowing that if I want to buy something online, chances are I can find it on Amazon. There are still a few apps that aren't on the App Store, that I wish were.

**Comment by njvack** [comment]
The main reason to not use the MAS would be to avoid dealing with sandboxing restrictions. Consumer apps are, generally, a better fit there than developer apps which often want to talk to areas of your filesystem (say, `/private/tmp/mysql.sock`) or do other things that are prohibited by the sandbox.

**Comment by plessbd** [comment]
Having a DMG might still be wanted, but it is not needed for this to work with homebrew. a `cask`, sequel-act already exists
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/sequel-ace.rb
just have to run it.
Might want to update the readme to point this out.  Having a DMG might still be wanted, but it is not needed for this to work with homebrew

**Comment by Jason-Morcos** [comment]
See #297 for the discussion of the cask.
As of now, the Sequel Ace team does not advise using the cask to install Sequel Ace. We aren't the ones controlling the cask. The only two official distributions of the app are the Mac App Store and GitHub releases.

**Comment by stychos** [comment]
Actually, Cask use our releases, but yes, we can't control it.

**Comment by JAAulde** [comment]
> Could you please hint me an idea or reason why would you prefer Brew instead of Mac AppStore? :)
> 
> Thanks!

Homebrew is preferable to me for a few reasons. The only one I'll annotate here is that it lets you keep different versions installed in parallel, with a quick way to jump between them as needed.

**Comment by Bytesmiths** [comment]
> On 2020-08-28, at 08:14, Jim Auldridge <notifications@github.com> wrote:
> 
> 
> Could you please hint me an idea or reason why would you prefer Brew instead of Mac AppStore? :)

The biggest reason is that homebrew is open-source and democratic, whereas the AppStore jerks both its customers and vendors around like fish on a line — when you're hooked, you do whatever they tell you to do.

Jan

:::: We waste enormous amounts of time and energy musing about the past and worrying about the future. These moments are hardly ever satisfying. -- Jon Kabat-Zinn <http://www.lmgtfy.com/?q=Jon+Kabat-Zinn> ::::
:::: Jan Steinman, EcoReality Co-op ::::
